TimeSeriesLifecycleActionsIT.testExplainFilters fails due to ILM retries

It's possible for the `testExplainFilters` test to fail due to ILM retrying the failed step and thus not being in the failed step. This causes an exception like:

```
java.lang.AssertionError: 
Expected: (map containing ["iaelhhxvxf-error"->ANYTHING] and map containing ["iaelhhxvxf-nonexistant-policy"->ANYTHING])
     but: map containing ["iaelhhxvxf-error"->ANYTHING] map was [<iaelhhxvxf-nonexistant-policy={index=iaelhhxvxf-nonexistant-policy, managed=true, policy=Xrtic, step_info={type=illegal_argument_exception, reason=policy [Xrtic] does not exist}}>]
	at __randomizedtesting.SeedInfo.seed([F284FECD14016E34:74B0734551DA485]:0)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
	at org.junit.Assert.assertThat(Assert.java:956)
	at org.junit.Assert.assertThat(Assert.java:923)
	at org.elasticsearch.xpack.ilm.TimeSeriesLifecycleActionsIT.lambda$testExplainFilters$47(TimeSeriesLifecycleActionsIT.java:866)
	at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:867)
	at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:840)
	at org.elasticsearch.xpack.ilm.TimeSeriesLifecycleActionsIT.testExplainFilters(TimeSeriesLifecycleActionsIT.java:853)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1750)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:938)
```

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+pull-request-2/10081/testReport/junit/org.elasticsearch.xpack.ilm/TimeSeriesLifecycleActionsIT/testExplainFilters/
https://gradle-enterprise.elastic.co/s/rgyylclrstq2o

The test should use a non-retryable error to simulate a policy in error, so that ILM doesn't auto-retry it and move the index out of the ERROR step.

--- a/x-pack/plugin/ilm/qa/multi-node/src/test/java/org/elasticsearch/xpack/ilm/TimeSeriesLifecycleActionsIT.java
+++ b/x-pack/plugin/ilm/qa/multi-node/src/test/java/org/elasticsearch/xpack/ilm/TimeSeriesLifecycleActionsIT.java
@@ -835,6 +835,23 @@ public class TimeSeriesLifecycleActionsIT extends ESRestTestCase {
 
         createFullPolicy(TimeValue.ZERO);
 
+        {
+            // Create a "shrink-only-policy"
+            Map<String, LifecycleAction> warmActions = new HashMap<>();
+            warmActions.put(ShrinkAction.NAME, new ShrinkAction(17));
+            Map<String, Phase> phases = new HashMap<>();
+            phases.put("warm", new Phase("warm", TimeValue.ZERO, warmActions));
+            LifecyclePolicy lifecyclePolicy = new LifecyclePolicy("shrink-only-policy", phases);
+            // PUT policy
+            XContentBuilder builder = jsonBuilder();
+            lifecyclePolicy.toXContent(builder, null);
+            final StringEntity entity = new StringEntity(
+                "{ \"policy\":" + Strings.toString(builder) + "}", ContentType.APPLICATION_JSON);
+            Request request = new Request("PUT", "_ilm/policy/shrink-only-policy");
+            request.setEntity(entity);
+            assertOK(client().performRequest(request));
+        }
+
         createIndexWithSettings(goodIndex, Settings.builder()
             .put(RolloverAction.LIFECYCLE_ROLLOVER_ALIAS, "alias")
             .put(IndexMetaData.SETTING_NUMBER_OF_REPLICAS, 0)
@@ -842,7 +859,7 @@ public class TimeSeriesLifecycleActionsIT extends ESRestTestCase {
         );
         createIndexWithSettingsNoAlias(errorIndex, Settings.builder()
             .put(IndexMetaData.SETTING_NUMBER_OF_REPLICAS, 0)
-            .put(LifecycleSettings.LIFECYCLE_NAME, policy)
+            .put(LifecycleSettings.LIFECYCLE_NAME, "shrink-only-policy")
         );
         createIndexWithSettingsNoAlias(nonexistantPolicyIndex, Settings.builder()
             .put(IndexMetaData.SETTING_NUMBER_OF_REPLICAS, 0)

